src/storage/tuple.cpp

```diff
@@ -59,6 +59,10 @@ void Tuple::SetValue(const oid_t column_offset, const type::Value &value,
 
   // Skip casting if type is same
   if (type == value.GetTypeId()) {
+    if ((type == type::Type::VARCHAR || type == type::Type::VARBINARY)
+        && value.GetLength() > column_length) {
+      throw peloton::IncompatibleTypeException(type, " data is too long");
```

Comment:
I don't think we want to throw `IncompatibleTypeException` here. Is there another exception type we can throw instead? What does Postgres do?